service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
+                    try {
+                        new_data = co_await service::calculate_tablet_replicas_for_new_rf(
+                                repl_strategy, table, tmptr, new_rf_per_dc_map);
+                    } catch(const std::exception& e) {
+                        new_data.status = tablet_replica_calculation_status::other;
+                        error = e.what();
+                        rtlogger.error("{}", error);
+                    }
+
+                    if (new_data.status != tablet_replica_calculation_status::success) {
+                        rtlogger.error("Couldn't process global_topology_request::keyspace_rf_change, status: {}, rf per dc map: {}",
+                                        new_data.status, new_data.new_tablet_map);
+                        updates.clear(); // remove all tablets mutations ...
+                        break;           // ... and only create mutations deleting the global req
+                    }
+
+                    auto tablet_id = new_data.new_tablet_map.first_tablet();
+                    while (true) {
+                        auto& tablet_info = new_data.new_tablet_map.get_tablet_info(tablet_id);
+                        auto last_token = new_data.new_tablet_map.get_last_token(tablet_id);
+                        updates.emplace_back(replica::tablet_mutation_builder(guard.write_timestamp(), table->id())
+                                                 .set_new_replicas(last_token, tablet_info.replicas)
+                                                 .set_stage(last_token, locator::tablet_transition_stage::allow_write_both_read_old)
+                                                 .set_transition(last_token, locator::tablet_transition_kind::rf_change)
+                                                 .build());
+
+                        if (auto next_tablet = new_data.new_tablet_map.next_tablet(tablet_id); next_tablet.has_value()) {
+                            tablet_id = *next_tablet;
+                        } else {
+                            break;
+                        }
+                    }
+                }
+
+                updates.push_back(canonical_mutation(topology_mutation_builder(guard.write_timestamp())
+                                                             .set_transition_state(topology::transition_state::tablet_migration)
+                                                             .set_version(_topo_sm._topology.version + 1)
+                                                             .del_global_topology_request()
+                                                             .del_global_topology_request_id()
+                                                             .build()));
+                updates.push_back(canonical_mutation(topology_request_tracking_mutation_builder(utils::UUID{req_uuid})
+                                                             .done(error)
```

Comment:
As I wrote above, `error` is not set on all paths and it's unclear if it's intentional or not.
I'd call it `status` rather than `error` and document its semantics for this change and for the future generations.

Reply:
ping

Reply:
Now `error` is only set when an actual error happens, and status codes have been removed.
As for renaming, `error` is the actual name of the column in the table that tracks requests, so I'd leave it like that.